### ISSUE TITLE
Allowing custom attributes to prevent unused parameter flagging for methods exported via Xamarin runtimes

### ISSUE THREAD

**Issue by Therzok** [dotnet/roslyn#34719]
**Version Used**: 
3.0.0-beta4-19157-04

**Steps to Reproduce**:

1. Create a Xamarin.iOS project, for example
2. Export some method to native that's not used in managed.

```csharp
	public class MyCustomButton : UIButton
	{
		[Export("performClick:")]
		void Click(NSObject o)
		{
		}
	}
```

3. Observe unused code diagnostics are reported

**Expected Behavior**:
Unused code diagnostics don't flag scopes marked with `[Foundation.ExportAttribute]`.

**Actual Behavior**:
We see members marked as unused, and they might get removed on bulk operations like fix all.

Attributes that need to be inserted somehow [here](http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.cs,158): `Java.Interop.ExportAttribute`, `Foundation.ExportAttribute`

How can that be done?

**Comment by Therzok** [comment]
cc @mavasani

**Comment by mavasani** [comment]
You can rename the parameter names to ‘_’, ‘_1’, ‘_2’, etc. to avoid seeing these violations.

**Comment by Therzok** [comment]
@mavasani that prevents parameters from being flagged, but not the method.

We need a mechanism to inject the two attributes into the set of attributes that skip analysis for the method, otherwise the analyzer will have destructive effects on user code.

In the sample above, we tell the runtime to create a handler in native that will be called when the iOS runtime sends a click message to a button. The code fixer will remove code, and that will not cause a build-time error.

**Comment by CyrusNajmabadi** [comment]
Closing out as not planned.
